### PR TITLE
Black-ify entire source base

### PR DESCRIPTION
**Pull request by james-smith-za**
Had to tweak the config files a bit to get black, isort and flake8 to play nicely together, but it seems that they are all more or less happy now.

### CODE REVIEW

File: tests/test_server.py [hunk 1]
```diff
@@ -241,249 +258,312 @@ async def test_sockets_not_started() -> None:
...
-    await check_reply(reader, [
-        b'!halt[11] ok\n',
-        br'#disconnect server\_shutting\_down' + b'\n',
-        b''])   # Empty string indicates EOF
+    server: DummyServer, reader: asyncio.StreamReader, writer: asyncio.StreamWriter
+) -> None:
+    writer.write(b"?halt[11]\n")
+    await check_reply(
+        reader, [b"!halt[11] ok\n", rb"#disconnect server\_shutting\_down" + b"\n", b""]
+    )  # Empty string indicates EOF
```
Comment (bmerry) [inline tests/test_server.py:435]: This comment is no longer on the same line as the empty string. I think put the arguments as one per line again, with the comment on the line with the empty string.

File: tests/test_server.py [hunk 2]
```diff
@@ -241,249 +258,312 @@ async def test_sockets_not_started() -> None:
...
+    writer.write(b"?wait[1]\n?halt[2]\n")
+    await check_reply(
+        reader,
+        [
+            b"!halt[2] ok\n",
+            b"!wait[1] fail request\\_cancelled\n",
+            b"#disconnect server\\_shutting\\_down\n",
+            b"",
+        ],
+    )  # Empty line indicates EOF
```
Comment (bmerry) [inline tests/test_server.py:451]: Move this comment two lines up, next to the empty string.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -16,6 +16,10 @@ repos:
     rev: 5.10.1
     hooks:
       - id: isort
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
```
Comment (bmerry) [inline .pre-commit-config.yaml:20]: Why not the latest (22.8.0)?

**Review by bmerry — CHANGES_REQUESTED** [review]
That went better than I was expecting - there are still a few places where its mindless approach of flowing each statement independently leads to related pieces of code being formatted inconsistently, but overall I think it's a good improvement.

File: .flake8 [hunk 4]
```diff
@@ -1,3 +1,3 @@
 [flake8]
-max-line-length = 100
+max-line-length = 120
```
[the remark this answers is not in the document]
Comment (james-smith-za) [reply .flake8:2]: I did, but `tests/test_client.py:749` doesn't get adjusted by `black`, I suspect due to the fact that the comment on the end (`: type: ignore`) doesn't get pushed on to the next lines for reasons that are probably obvious.

So it seemed to me that the easiest solution was to give flake8 a little bit of ullage space compared with black's configuration.

[the remark this answers is not in the document]
Comment (bmerry) [reply .flake8:2]: Really? If I start from master and run `black --max-line-length 100 tests/test_client.py` then it does something sensible. Did you perhaps start with a line length of 120 and re-ran black with a length of 100? That might cause it to first flatten the whole thing into one line, then be unable to split it correctly again.

[the remark this answers is not in the document]
Comment (james-smith-za) [reply .flake8:2]: In retrospect that's probably what happened, because I did start with 120 I think. I had `black` configured like in `katgpucbf`, but it was fighting with `isort` which was configured with 100, it took embarassingly long to figure out that this was the problem. Once I then changed black to 100, this was the result. Let me try again.

**Comment by james-smith-za** [comment]
I *think* the latest commit addresses all your comments.

File: pyproject.toml [hunk 5]
```diff
@@ -7,19 +7,12 @@ line_length = 100
...
-		| \.venv
-		| _build
-		| buck-out
-		| build
-		| dist
+      \.pyi?$
+    | setup.py
+    | src
+    | test
+    | examples
```
Comment (bmerry) [inline pyproject.toml:16]: This looks like it'll run on any file with `src` anywhere in its filename, even if it doesn't have the right extension, or isn't part of the top-level `src` tree.

What happens if you just remove this setting entirely (the default is `(\.pyi?|\.ipynb)$`)? According to the docs, if you don't specify an `exclude` then it will automatically honour `.gitignore`, which should prevent it reformatting anything that isn't part of the source.

**Review by bmerry — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 5 above)
Comment (james-smith-za) [reply pyproject.toml]: That seems to have worked.

I think if I merge, I'm going to do a rebase and squash everything into a single commit, so that I can add a little file saying which commit to ignore for git blame:
https://black.readthedocs.io/en/stable/guides/introducing_black_to_your_project.html#avoiding-ruining-git-blame

**Review by bmerry — APPROVED** [review]
LGTM, and I agree with your suggestion about squashing.
